File: src/test/resources/oracle/commits/rm-action-test-288d1a960b5edef78540921f78809cafb3ef3ff6/typescript/CustomerProfile.ts
import { Address } from "./Address";

export class CustomerProfile {
    private displayName: string;
    private address: Address;
    private loyaltyPoints: number;

    constructor(fullName: string, street: string, city: string, postalCode: string, loyaltyPoints: number) {
        this.displayName = fullName;
        this.address = new Address(street, city, postalCode);
        this.loyaltyPoints = loyaltyPoints;
    }

    mailingLabel(): string {
        return `${this.displayName}\n${this.address.format()}`;
    }

    canReceivePromotion(emailOptedIn: boolean, bouncedEmail: boolean, purchaseCount: number): boolean {
        if (!emailOptedIn || bouncedEmail || purchaseCount <= 0) {
            return false;
        }
        return true;
    }

    addPurchase(amountCents: number): void {
        const earnedPoints = Math.floor(amountCents / 100);
        this.loyaltyPoints += earnedPoints;
    }

    getLoyaltyPoints(): number {
        return this.loyaltyPoints;
    }
}

File: src/test/resources/oracle/commits/rm-action-test-b65346600e43fad70b02f037dda796f86015f0c6/python/customer_profile.py
from address import Address


class CustomerProfile:
    def __init__(self, full_name, street, city, postal_code, loyalty_points):
        self._display_name = full_name
        self._address = Address(street, city, postal_code)
        self._loyalty_points = loyalty_points

    def mailing_label(self):
        return f"{self._display_name}\n{self._address.format()}"

    def can_receive_promotion(self, email_opted_in, bounced_email, purchase_count):
        if not email_opted_in or bounced_email or purchase_count <= 0:
            return False
        return True

    def add_purchase(self, amount_cents):
        earned_points = amount_cents // 100
        self._loyalty_points += earned_points

    def get_loyalty_points(self):
        return self._loyalty_points
